backend/src/main/java/hu/flowacademy/worksheet/controller/UserController.java

```diff
@@ -44,9 +47,21 @@ public AccessTokenResponse login(@RequestBody UserOperationDTO userOperationDTO)
         return keycloakClientService.login(userOperationDTO.getEmail(), userOperationDTO.getPassword());
     }
 
+    private final PagingProperties pagingProperties;
+
+    @RolesAllowed("admin")
     @GetMapping("/users")
-    @ResponseStatus(HttpStatus.FOUND)
-    public List<User> findUserByNameOrEmail(@RequestParam(value = "q") String q) {
-        return userService.findUserByNameAndEmail(q);
+    public List<User> getRegistrations(@RequestParam(value = "page", required = false) Optional<Integer> page,
```

Comment:
A `@RequestParam(value = "q") String q`-t tartsuk meg, csak tedd ezt is opcionálissá és küld át szűrésre. Bővebben a következő commentben írok.